src/objects/PieceCondition.ts

```diff
@@ -0,0 +1,11 @@
+import { Field, Struct, Poseidon } from 'snarkyjs';
+
+// If this is always going to be an exact copy of stats then maybe we can condense the 2
+export class PieceCondition extends Struct({
+  health: Field,
+  movement: Field,
+}) {
```

Comment:
The GamePiece will definitely have its own `health` since that can be reduced during the game prior to the piece dying. Further I'd say it's a good call for the piece to have its own `maxHealth` as well since part of the value of the idea of a `PlayerUnit` (your special version of that Unit you carry around between games) is that it gains experience and maybe some small buffs over time. Perhaps something like +5 to `maxHealth` or +10 `movementSpeed` or something. Irrelevant for now but here is an example of some upgrades from WH which could inspire the concept:

<img width="404" alt="Screen Shot 2023-05-02 at 7 11 43 PM" src="https://user-images.githubusercontent.com/8811423/235804632-d54799c6-5098-483d-ac15-0bda1e8f300d.png">

Reply:
A piece has a `condition` but the unit that it implements has `stats`.  They are the same shape, but condition can be edited during the game and unit stats cannot.  So unit stats could change game to game.